modules/storage/src/main/kotlin/edu/stanford/spezi/modules/storage/key/LocalKeyValueStorage.kt

```diff
@@ -0,0 +1,46 @@
+package edu.stanford.spezi.modules.storage.key
+
+import android.content.Context
+import androidx.datastore.core.DataStore
+import androidx.datastore.preferences.core.Preferences
+import androidx.datastore.preferences.core.edit
+import androidx.datastore.preferences.preferencesDataStore
+import dagger.hilt.android.qualifiers.ApplicationContext
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.first
+import kotlinx.coroutines.flow.map
+import javax.inject.Inject
+
+class LocalKeyValueStorage @Inject constructor(
+    @ApplicationContext context: Context,
+) : KeyValueStorage {
+    companion object {
+        private val Context.dataStore: DataStore<Preferences> by preferencesDataStore(
+            name = "spezi_preferences"
+        )
+    }
+
+    private val dataStore = context.dataStore
+
+    override suspend fun <T : Any> saveData(key: PreferenceKey<T>, data: T) {
+        dataStore.edit { preferences ->
+            preferences[key.key] = data
+        }
+    }
+
+    override fun <T> readData(key: PreferenceKey<T>): Flow<T?> {
+        return dataStore.data.map { preferences ->
```

Comment:
In general all `dataStore.data` should be followed by `dataStore.data..catch { emit(emptyPreferences()) }` to avoid throwing and using an empty preferences file in case something went wrong when accessing the file